pkg/store/bucket_test.go

```diff
@@ -303,3 +305,27 @@ func (s *testStoreSeriesServer) Send(r *storepb.SeriesResponse) error {
 func (s *testStoreSeriesServer) Context() context.Context {
 	return s.ctx
 }
+
+func uploadDir(t testing.TB, ctx context.Context, bkt shipper.Bucket, id ulid.ULID, dir string) error {
+	err := filepath.Walk(dir, func(src string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if fi.IsDir() {
+			return nil
+		}
+
+		target := filepath.Join(id.String(), strings.TrimPrefix(src, dir))
+		return bkt.Upload(ctx, src, target)
+	})
+	if err == nil {
+		return nil
+	}
+
+	// We don't want to leave partially uploaded directories behind. Cleanup everything related to it
+	// and use a uncanceled context.
+	if err2 := bkt.Delete(ctx, dir); err2 != nil {
+		t.Log("cleanup failed; partial data may be left behind. dir: %s. Err: %v", dir, err2)
+	}
+	return err
+}
```

Comment:
Could it be useful to have backend-agnostic helpers like this in `pkg/objstore`?

Reply:
As discussed offline, this function is block related and also used only here. Similar one is used in shipper prod code, but it has additional metrics. Not sure where and if it is worth to migrate this function.